src/Dataset.ts

```diff
@@ -577,6 +577,101 @@ export default class Dataset {
     }
     return this._allPrefixes;
   }
+
+  private async apiNDEDatasetRegister(method: "validate" | "submit", next?: (shaclReport: n3.Store) => void) {
```

Comment:
can we find a better name for `next`? Based on the docs, it seems it should be called something like `handleShaclReport` or something?

Reply:
I used `next` because of my experience using `express` framework  ;-) I have changed it to your suggestion.

Reply:
Come to think of it: it would be better to return a `Promise<n3.Store>` and do not use a callback at all.